lib/taskana-core/src/main/java/pro/taskana/TaskanaConfiguration.java

```diff
@@ -0,0 +1,696 @@
+      }
+      loadProperties(propertiesFile);
+      configureAnnotatedFields(this, separator, properties);
+      roleMap = configureRoles(separator, properties, shouldUseLowerCaseForAccessIds());
+      classificationCategoriesByTypeMap =
+          configureClassificationCategoriesForType(properties, classificationTypes);
+      return this;
+    }
+
+    /**
+     * This method creates a PooledDataSource, if the needed properties are provided.
+     *
+     * @param driver the name of the jdbc driver
+     * @param jdbcUrl the url to which the jdbc driver connects
+     * @param username the user name for database access
+     * @param password the password for database access
+     * @return DataSource
+     */
+    public static DataSource createDatasource(
+        String driver, String jdbcUrl, String username, String password) {
+      return new PooledDataSource(driver, jdbcUrl, username, password);
+    }
+
+    private String initSchemaName(String schemaName) {
+      String schemaNameTmp;
+      if (schemaName != null && !schemaName.isEmpty()) {
+        schemaNameTmp = schemaName;
+      } else {
+        throw new SystemException("SchemaName can't be null or empty");
+      }
+
+      try (Connection connection = dataSource.getConnection()) {
+        String databaseProductId = DB.getDatabaseProductId(connection);
+        if (DB.isPostgres(databaseProductId)) {
+          schemaNameTmp = schemaNameTmp.toLowerCase();
+        } else {
+          schemaNameTmp = schemaNameTmp.toUpperCase();
+        }
+      } catch (SQLException ex) {
+        LOGGER.error("Caught exception when attempting to initialize the schema name", ex);
+      }
+
+      if (LOGGER.isDebugEnabled()) {
+        LOGGER.debug("Using schema name {}", schemaNameTmp);
+      }
+      return schemaNameTmp;
+    }
+
+    private DataSource createDefaultDataSource() {
+      String driverClass = "org.h2.Driver";
+      String jdbcUrl =
+          "jdbc:h2:mem:taskana;NON_KEYWORDS=KEY,VALUE;"
+              + "IGNORECASE=TRUE;LOCK_MODE=0;"
+              + "INIT=CREATE SCHEMA IF NOT EXISTS TASKANA\\;"
+              + "SET COLLATION DEFAULT_de_DE";
+      String username = "sa";
+      String password = "sa";
+      LOGGER.info(
+          "No datasource is provided. An in-memory db is used: " + "'{}', '{}', '{}', '{}'",
+          driverClass,
+          jdbcUrl,
+          username,
+          password);
+      return createDatasource(driverClass, jdbcUrl, username, password);
+    }
+
+    private void loadProperties(String propertiesFile) {
+      Properties props = new Properties();
+      try (InputStream stream =
+          FileLoaderUtil.openFileFromClasspathOrSystem(
+              propertiesFile, TaskanaConfiguration.class)) {
+        props.load(stream);
+      } catch (IOException e) {
+        throw new SystemException(
+            "internal System error when processing properties file " + propertiesFile, e);
+      }
+      this.properties = props;
```

Comment:
replace this with an immutable map. So that we don't have to convert that Map in the TaskanaEngine constructor